File: BinaryTree and Divide and Conquer/le101_symmetricTree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root == None:
            return True
        return self.symHelper(root.left, root.right)
    
    def symHelper(self, one, two):
        if one == None and two == None:
            return True
        elif one == None or two == None:
            return False
        else:
            if one.val == two.val:
                return self.symHelper(one.left, two. right) and self.symHelper(one.right, two.left)
            else:
                return False

File: JavaPartI/le101_symmetricTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if (root == null) {
            return true;
        }
        return symHelper(root.left, root.right);
    }
    
    private boolean symHelper(TreeNode one, TreeNode two) {
        if (one == null && two == null) {
            return true;
        } else if (one == null || two == null) {
            return false;
        } else {
            if (one.val != two.val) {
                return false;
            }
            return symHelper(one.left, two.right) && symHelper(one.right, two.left);
        }
        
    }
}
